lib/native/auth.spec.ts

```diff
@@ -187,4 +188,104 @@ test('Default credentials provider create no bootstrap', async () => {
+        ],
+        customRoleArn: "arn:us-east-1:totally4real",
+        tlsContext: tlsCtx,
+        bootstrap: new native_io.ClientBootstrap(),
+    };
+
+    const credentials_provider = native.AwsCredentialsProvider.newCognito(config);
+
+    expect(credentials_provider);
+});
+
+const AWS_TESTING_COGNITO_IDENTITY : string = process.env.AWS_TESTING_COGNITO_IDENTITY ?? "";
+
+function hasCognitoTestEnvironment() {
+    return AWS_TESTING_COGNITO_IDENTITY !== "";
+}
+
+const conditional_test = (condition : boolean) => condition ? it : it.skip;
+
+async function do_body_request_signing(provider: native.AwsCredentialsProvider) {
+    const signing_config: native.AwsSigningConfig = {
+        algorithm: native.AwsSigningAlgorithm.SigV4,
+        signature_type: native.AwsSignatureType.HttpRequestViaHeaders,
+        provider: provider,
+        region: SIGV4TEST_REGION,
+        service: SIGV4TEST_SERVICE,
+        date: new Date(DATE_STR),
+        signed_body_header: native.AwsSignedBodyHeaderType.None,
+    };
+    let stream = new PassThrough();
+    let body_stream;
+    stream.write("test");
+    stream.end(() => {
+        body_stream = new InputStream(stream);
+    });
+    let http_request = new native_http.HttpRequest(
+        SIGV4TEST_METHOD,
+        SIGV4TEST_PATH,
+        new native_http.HttpHeaders(SIGV4TEST_UNSIGNED_HEADERS),
+        body_stream);
+
+    return await aws_sign_request(http_request, signing_config);
+}
+
+conditional_test(hasCognitoTestEnvironment())('Cognito credentials provider usage success - signing', async () => {
+    let config : native.CognitoCredentialsProviderConfig = {
+        endpoint: "cognito-identity.us-east-1.amazonaws.com",
+        identity: AWS_TESTING_COGNITO_IDENTITY
+    };
+
+    const credentials_provider = native.AwsCredentialsProvider.newCognito(config);
+
+    expect(credentials_provider);
+
+    const signing_result = await do_body_request_signing(credentials_provider);
+
+    expect(signing_result.method).toBe(SIGV4TEST_METHOD);
+    expect(signing_result.path).toBe(SIGV4TEST_PATH);
+});
+
+conditional_test(hasCognitoTestEnvironment() && ProxyConfig.is_valid())('Cognito credentials provider through http proxy usage success - signing', async () => {
+    let proxyOptions: native_http.HttpProxyOptions = ProxyConfig.create_http_proxy_options_from_environment(
+        ProxyTestType.TUNNELING_HTTPS, native_http.HttpProxyAuthenticationType.None);
+
+    let config : native.CognitoCredentialsProviderConfig = {
+        endpoint: "cognito-identity.us-east-1.amazonaws.com",
+        identity: AWS_TESTING_COGNITO_IDENTITY,
+        httpProxyOptions: proxyOptions
+    };
+
+    const credentials_provider = native.AwsCredentialsProvider.newCognito(config);
+
+    expect(credentials_provider);
+
+    const signing_result = await do_body_request_signing(credentials_provider);
+
+    expect(signing_result.method).toBe(SIGV4TEST_METHOD);
+    expect(signing_result.path).toBe(SIGV4TEST_PATH);
```

Comment:
I believe this test never runs on CI. The only place we set up the proxy config are the codebuild job, which simply runs `npm test` [here](https://github.com/awslabs/aws-crt-nodejs/blob/main/codebuild/linux-integration-tests.sh). So, it will not go through the builder to setup the cognito config.

Maybe update the codebuild job to use builder instead?

Reply:
I've updated the environment setup script in s3 to also initialize the cognito environment variable.

Reply:
Looks like it breaks all the ci that has not updated the builder version...
https://github.com/awslabs/aws-c-http/actions/runs/3101997039/jobs/5023883935#step:2:4009

Reply:
That seems like a problem with the stuff invoking the downstream builds.  Given builder versioning, you shouldn't be doing a downstream build of a target using *your* builder version, you should be using the downstream target's builder version.